src/middlewared/middlewared/plugins/apps/schema_utils.py

```diff
@@ -0,0 +1,152 @@
+import itertools
+from typing import Tuple, Union
+
+from middlewared.service import ValidationErrors
+from middlewared.schema import (
+    Attribute, Bool, Cron, Dict, Dir, File, HostPath, Int, IPAddr, List, NOT_PROVIDED, Path, Str, URI,
+)
+from middlewared.validators import Match, Range, validate_schema
+
+
+CONTEXT_KEY_NAME = 'ix_context'
+RESERVED_NAMES = [
+    ('ix_certificates', dict),
+    ('ix_certificateAuthorities', dict),
+    ('ix_externalInterfacesConfiguration', list),
+    ('ix_externalInterfacesConfigurationNames', list),
```

Comment:
Should we leave a todo here to update/fix casing when those are implemented?

Reply:
Yes a todo would be nice, this list would be cleared when we implement ref support which is next btw - will push a todo